modules/mono/editor/bindings_generator.cpp

```diff
@@ -2248,15 +2260,48 @@ Error BindingsGenerator::_generate_cs_signal(const BindingsGenerator::TypeInterf
 			p_output.append("\")]");
 		}
 
-		String delegate_name = p_isignal.proxy_name;
-		delegate_name += "EventHandler"; // Delegate name is [SignalName]EventHandler
+		bool is_parameterless = p_isignal.arguments.size() == 0;
 
-		// Generate delegate
-		p_output.append(MEMBER_BEGIN "public delegate void ");
-		p_output.append(delegate_name);
-		p_output.append("(");
-		p_output.append(arguments_sig);
-		p_output.append(");\n");
+		// Delegate name is [SignalName]EventHandler
+		String delegate_name = is_parameterless ? "Action" : p_isignal.proxy_name + "EventHandler";
+
+		if (!is_parameterless) {
+			// Generate delegate
+			p_output.append(MEMBER_BEGIN "public delegate void ");
+			p_output.append(delegate_name);
+			p_output.append("(");
+			p_output.append(arguments_sig);
+			p_output.append(");\n");
+
+			// Generate Callable trampoline for the delegate
+			p_output << MEMBER_BEGIN "[System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]\n"
```

Comment:
I'm not sure, tbh. I added it just in case because I don't see any other instance private method in Godot Objects, only static ones.

Reply:
Since the purpose of the attribute with the state set to Never is to hide the member in IDEs, and this member is already private (which means IDEs will hide it since it's not accessible) I don't think the attribute is needed. Private members are also inaccessible from derived types so they also shouldn't show up in an IDE.

Or am I misunderstanding what you are saying?

Reply:
What I mean is that I think I remember Visual Studio was suggesting me private members long ago. This may have been changed since then, or it may have never been the case and my memory is just playing tricks on me.

Reply:
Oh, well that would be a Visual Studio bug because there's no reason to suggest a member that you can't access, or maybe you're just misremembering. I don't use Visual Studio but I can confirm that in VSCode private members are not suggested by IntelliSense so I would expect the same behavior in Visual Studio.

Reply:
All right, I'll remove it. We could always add it back if needed.